File: festival-kotlin-web/src/main/java/cn/bdqfork/web/WebApplication.java
package cn.bdqfork.web;

/**
 * @author bdq
 * @since 2020/2/17
 */
public class WebApplication {
    public void run(Class<?> clazz) {
        String scanPath = clazz.getPackage().getName();
        try {
            new WebApplicationContext(scanPath).start();
        } catch (Exception e) {
            throw new IllegalStateException(e);
        }
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/WebApplication.kt
package cn.bdqfork.kotlin.web

/**
 * @author bdq
 * @since 2020/2/17
 */
class WebApplication {

    companion object {
        fun run(clazz: Class<*>) {
            val scanPath = clazz.getPackage().name
            try {
                WebApplicationContext(scanPath).start()
            } catch (e: Exception) {
                throw IllegalStateException(e)
            }
        }
    }
}